service/controller/resource/deployment/releasecomponents.go

```diff
@@ -0,0 +1,30 @@
+package deployment
+
+import (
+	"github.com/giantswarm/kvm-operator/service/controller/key"
+
+	releasev1alpha1 "github.com/giantswarm/apiextensions/v3/pkg/apis/release/v1alpha1"
+)
+
+var keyComponents = []string{"calico", "containerlinux", "etcd", "kubernetes"}
+
+func addKeyComponentsAnnotations(annotations map[string]string, release *releasev1alpha1.Release) {
```

Comment:
I personally would prefer a pure function here. This would require you to create a copy of the annotations map, add the component versions to the copy, and return it from the function. I'm not blocking though if you think this way is better.

Reply:
Maps are just pointers in Go so no copying is happening in this function. That's why it also doesn't also return the map, it just edits the original one. But I can still change the function if you think it's clearer. My reasoning was that it should only get the input it needs and it's easier to follow what's happening with the Annotations in the deployments.

Reply:
Updated